packages/wonder-blocks-tooltip/components/tooltip-anchor.test.js

```diff
@@ -0,0 +1,560 @@
+import {mount} from "enzyme";
+
+import TooltipAnchor from "./tooltip-anchor.js";
+import TooltipPortalMounter from "./tooltip-portal-mounter.js";
+
+// Some helpers for chaining event handlings.
+const doEvent = (ref, event, then) => {
+    ref.dispatchEvent(event);
+    if (then) {
+        setTimeout(then, 0);
+    }
+};
+
+describe("TooltipAnchor", () => {
+    test("on mount, subscribes to focus and hover events", () => {
+        // Arrange
+        const getFakeTooltipPortalMounter = (active) =>
+            (((
+                <div id="anchor">{active ? "true" : "false"}</div>
+            ): any): React.Element<typeof TooltipPortalMounter>);
+        const nodes = (
+            <div>
+                <TooltipAnchor anchorRef={() => {}}>
+                    {getFakeTooltipPortalMounter}
+                </TooltipAnchor>
+            </div>
+        );
+        const addEventListenerSpy = jest.spyOn(
+            HTMLElement.prototype,
+            "addEventListener",
+        );
+        addEventListenerSpy.mockClear();
+
+        // Act
+        mount(nodes);
+
+        // Assert
+        expect(addEventListenerSpy).toHaveBeenCalledWith(
+            "focusin",
+            expect.any(Function),
+        );
+        expect(addEventListenerSpy).toHaveBeenCalledWith(
+            "focusout",
+            expect.any(Function),
+        );
+        expect(addEventListenerSpy).toHaveBeenCalledWith(
+            "mouseenter",
+            expect.any(Function),
+        );
+        expect(addEventListenerSpy).toHaveBeenCalledWith(
+            "mouseleave",
+            expect.any(Function),
+        );
+    });
+
+    test("on unmount, unsubscribes from focus and hover events", () => {
+        // Arrange
+        const getFakeTooltipPortalMounter = (active) =>
+            (((
+                <div id="anchor">{active ? "true" : "false"}</div>
+            ): any): React.Element<typeof TooltipPortalMounter>);
+        const nodes = (
+            <div>
+                <TooltipAnchor anchorRef={() => {}}>
+                    {getFakeTooltipPortalMounter}
+                </TooltipAnchor>
+            </div>
+        );
+        const removeEventListenerSpy = jest.spyOn(
+            HTMLElement.prototype,
+            "removeEventListener",
+        );
+        removeEventListenerSpy.mockClear();
+        const wrapper = mount(nodes);
+
+        // Act
+        wrapper.unmount();
+
+        // Assert
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "focusin",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "focusout",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "mouseenter",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "mouseleave",
+            expect.any(Function),
+        );
```

Comment:
I wish there a was a way to check what handlers are currently subscribed to events on an element.

Reply:
Indeed.